controllers/cloud.redhat.com/metrics.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	totalSuccessfulPoolReservationsCountMetrics = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "total_successful_pool_reservation_counter",
+			Help: "Total successful reservations from each pool",
+		},
+		[]string{"pool"},
+	)
+
+	totalFailedPoolReservationsCountMetrics = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "total_failed_pool_reservation_counter",
+			Help: "Total failed reservations from each pool",
+		},
+		[]string{"pool"},
+	)
+
+	averageRequestedDurationMetrics = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "average_duration_for_namespace_reservations_in_hours",
+			Help:    "Average duration for namespace reservations (In hours)",
+			Buckets: []float64{1, 2, 4, 8, 24, 48, 168, 336},
```

Comment:
I really like the bucket selection here. Might be worth adding a comment that an `Inf+` bucket will be made implicitly by the prom library.